### ISSUE TITLE
Can't move window to second screen

### ISSUE THREAD

**Issue by chifandeyu** [Bringer-of-Light/Qt-Nice-Frameless-Window#50]
Hold down the left mouse button press and hold the left mouse button to move the window to the second screen the release of the mouse, and then click on the title bar to move the window, can not move

**Comment by yeahitsjan** [comment]
I can reproduce this. @chifandeyu you need to resize the window, then it shoud work again.

But its pretty annoying. Also the "titlebar" widget gets a bit wider then, when resizing its back to normal...

**Comment by SimonBuxx** [comment]
I can also reproduce this, happens every time I move the window from one screen to the other.

**Comment by theoyjy** [comment]
yep, even though I have two exact same screen, this size of the widget is expanded when moving the widget across screens, and actual position of titlebar is below what it is showing

**Comment by LochlanPerry-Scott** [comment]
Howdy all,
@chifandeyu
I have had a quick look into this and it seems like this might be caused by the Resolution or the Scale of the monitor.
I have a 3 monitor setup with a laptop, all work fine when I have no Scale (100%) added into my Windows Settings.
But when changing this to a higher scale (125%) on laptop or monitor ! can no longer move the window, resize works fine and so do button actions.. so I'm guessing this might be due to the calculation of the DPI support.
(as this is mainly used in the title bar positioning)

I may continue to troubleshoot this further, but here is a starting point.

**Comment by LochlanPerry-Scott** [comment]
Howdy again.
I found a solution to this one too, seems this is not an issue with the Nice-Frameless-Window, but how DPI is used in Qt itself.
I am using Qt6 at this moment.

Debugging the framelesswindow showed me that the issue was stemming from the 'mapFromGlobal(QPoint(x/dpr,y/dpr))' function used for mouse position calculation.

Looking into the High DPI Qt Docs.
https://doc.qt.io/qt-5/highdpi.html
(I know this is Qt5 Docs but im using Qt6.3 and this doc changed in v6.4) - using the below works anyway.
It shows that we may have to enable the '<application> -platform windows:dpiawareness=0,1,2', Command line argument for the main function.

QT goes into detail about how this works, but I personally am handling this via the code below before my main application function. 'main(int argc, char** argv)'

```
// Setup new argv value
std::vector<const char*> new_argv(argv, argv + argc);

// Pushback the arguments
new_argv.push_back("-platform");
new_argv.push_back("windows:dpiawareness=1");

// Calculate Difference in the argv values
int diff = new_argv.size() - argc;
argv = (char**)new_argv.data();			// Set argv value back to the new_argv (Convert back to char**, this is in the main())

// Add the Difference the the argc count
argc = argc + diff;

// Print to Debug to see the values
qDebug() << "You have entered " << argc << " arguments:" << "\n";
for (int i = 0; i < argc; ++i)
	qDebug() << argv[i] << "\n";
```

This prints out the below and successfully allows me to correctly use the Titlebar across multi-DPI screens.. from my little testing lol.
```
- The apps location .exe -  

-platform 

windows:dpiawareness=1 
```

Edit:
Turns out this is due to me setting the dpiawareness to System-DPI Aware.

Hope this helps your findings.

**Comment by dev-solo-game** [comment]
has this problem been fixed ?

**Comment by yeahitsjan** [comment]
@jackfong66 Not until now. The project seems dead.. I moved to [FramelessHelper](https://github.com/wangwenx190/framelesshelper).

**Comment by dev-solo-game** [comment]
> 

thx

**Comment by dev-solo-game** [comment]
> @jackfong66 Not until now. The project seems dead.. I moved to [FramelessHelper](https://github.com/wangwenx190/framelesshelper).
目前用这个方法可以暂时解决
void FramelessWindow::moveEvent(QMoveEvent*event)
{
    if (!m_currentScreen)
    {
        m_currentScreen = screen();
    } else if (m_currentScreen != screen())
    {
        m_currentScreen = screen();
        SetWindowPos((HWND) winId(), NULL, 0, 0, 0, 0,
                     SWP_NOMOVE | SWP_NOSIZE | SWP_NOZORDER | SWP_NOOWNERZORDER | SWP_FRAMECHANGED | SWP_NOACTIVATE);
    }
    QMainWindow::moveEvent(event);
}
